google/cloud/firestore_v1/stream_generator.py

```diff
@@ -14,21 +14,53 @@
 
 """Classes for iterating over stream results for the Google Cloud Firestore API.
 """
+from __future__ import annotations
 
-from collections import abc
+from typing import TYPE_CHECKING, Any, Generator, Optional, TypeVar
 
+from google.cloud.firestore_v1.query_profile import (
+    ExplainMetrics,
+    QueryExplainError,
+)
 
-class StreamGenerator(abc.Generator):
-    """Generator for the streamed results."""
+if TYPE_CHECKING:  # pragma: NO COVER
+    from google.cloud.firestore_v1.query_profile import ExplainOptions
 
-    def __init__(self, response_generator):
+
+T = TypeVar("T")
+
+
+class StreamGenerator(Generator[T, Any, None]):
+    """Generator for the streamed results.
+
+    Args:
+        response_generator (Generator[T, Any, Optional[ExplainMetrics]]):
+            The inner generator that yields the returned document in the stream.
+        explain_options
+            (Optional[:class:`~google.cloud.firestore_v1.query_profile.ExplainOptions`]):
+            Query profiling options for this stream request.
+    """
+
+    def __init__(
+        self,
+        response_generator: Generator[T, Any, Optional[ExplainMetrics]],
+        explain_options: Optional[ExplainOptions] = None,
+    ):
         self._generator = response_generator
+        self._explain_options = explain_options
+        self._explain_metrics = None
 
     def __iter__(self):
-        return self._generator
+        return self
 
     def __next__(self):
```

Comment:
nit: you can probably add `-> T` return annotation here, which might help with type checking

Reply:
done